### PR TITLE
Get utxo record details 

### LINKED ISSUE #568: Get utxo record details
**Brief of your idea**
Get utxo record infos including TotalUtxoRecord, LockedUtxoRecord and FrozenUtxoRecord

**What could be better if XuperChain using your idea**
It is beneficial to debug.

**Are there any side effects?**
None

**Additional information**
image

### PR DESCRIPTION
**Pull request by ToWorld**
## Description

What is the purpose of the change?
User can get more details about utxo records.

Fixes #568 

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## Brief of your solution

Fetch the infos such as totalUtxoRecord, LockedUtxoRecord and FrozenUtxoRecord from KVDB.

## How Has This Been Tested?

Regression Test

### CODE REVIEW

File: pb/xchain.proto [hunk 1]
```diff
@@ -951,3 +958,18 @@ message ModifyBlock {
...
+  string accountName = 3; 
+  UtxoRecord totalUtxoRecord = 4;
+  UtxoRecord lockedUtxoRecord = 5;
+  UtxoRecord frozenUtxoRecord = 6;
+  int64 displayCount = 7;
+  repeated string utxoItem = 8;  
+}
+
+message UtxoRecord {
+  string utxoCount = 1;
```
[the remark this answers is not in the document]
Comment (ToWorld) [reply pb/xchain.proto]: The maximum value of utxoCount is the total supply of utxo, so it's possible that utxoCount is bigger than max(int64).

File: utxo/utxo.go [hunk 2]
```diff
@@ -2332,3 +2332,68 @@ func (uv *UtxoVM) queryContractBannedStatus(contractName string) (bool, error) {
 	ctx.Release()
 	return false, nil
 }
+
+// QueryUtxoRecord query utxo record details
+func (uv *UtxoVM) QueryUtxoRecord(accountName string, displayCount int64) (*pb.UtxoRecordDetail, error) {
+	defaultUtxoRecordDetail := &pb.UtxoRecordDetail{
```
[the remark this answers is not in the document]
Comment (ToWorld) [reply utxo/utxo.go]: default value to be returned when error occurs.

[the remark this answers is not in the document]
Comment (ToWorld) [reply utxo/utxo.go]: default value is to be returned when error occurs.

File: pb/xchain.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (icexin) [reply pb/xchain.proto]: What's the meaning of `utxoAmount`?

File: cmd/cli/account.go [hunk 3]
```diff
@@ -18,14 +18,15 @@ func NewAccountCommand(cli *Cli) *cobra.Command {
 	c.cli = cli
 	c.cmd = &cobra.Command{
 		Use:   "account",
-		Short: "Operate an account or address: balance|new|newkeys|split.",
+		Short: "Operate an account or address: balance|new|newkeys|split|query-utxo-record.",
```
Comment (yucaowang) [inline cmd/cli/account.go]: keep command short and meaningful, for example list-utxo/query-utxo.

File: utxo/utxo.go [hunk 4]
```diff
@@ -2332,3 +2332,68 @@ func (uv *UtxoVM) queryContractBannedStatus(contractName string) (bool, error) {
...
+		Header: &pb.Header{},
+	}
+
+	addrPrefix := fmt.Sprintf("%s%s_", pb.UTXOTablePrefix, accountName)
+	it := uv.ldb.NewIteratorWithPrefix([]byte(addrPrefix))
```
Comment (yucaowang) [inline utxo/utxo.go]: Is iterator thread-safe for concurrent operations?

File: utxo/utxo.go [hunk 5]
```diff
@@ -2332,3 +2332,68 @@ func (uv *UtxoVM) queryContractBannedStatus(contractName string) (bool, error) {
...
+	lockedUtxoAmount := big.NewInt(0)
+	frozenUtxoCount := big.NewInt(0)
+	frozenUtxoAmount := big.NewInt(0)
+	utxoItems := []string{}
+
+	for it.Next() {
+		totalUtxoCount.Add(totalUtxoCount, big.NewInt(1))
+		key := append([]byte{}, it.Key()...)
+		if totalUtxoCount.Cmp(big.NewInt(displayCount)) <= 0 {
+			utxoItems = append(utxoItems, string(key))
```
Comment (yucaowang) [inline utxo/utxo.go]: So we only need utxo key in the utxo record list?

File: cmd/cli/account_utxo_records.go [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+			return c.queryUtxoRecords(ctx)
+		},
+	}
+	c.addFlags()
+	return c.cmd
+}
+
+func (c *QueryUtxoRecordsCommand) addFlags() {
+	c.cmd.Flags().StringVar(&c.addr, "address", "", "address")
+	c.cmd.Flags().Int64Var(&c.utxoItemNum, "N", 0, "utxo items to be displayed")
```
Comment (yucaowang) [inline cmd/cli/account_utxo_records.go]: command flags like "N" should be short for a meaningful word in xchain-cli, like in split utxo, N is short for "num".

File: cmd/cli/account_utxo_records.go [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+			ctx := context.TODO()
+			return c.queryUtxoRecords(ctx)
+		},
+	}
+	c.addFlags()
+	return c.cmd
+}
+
+func (c *QueryUtxoRecordsCommand) addFlags() {
+	c.cmd.Flags().StringVar(&c.addr, "address", "", "address")
```
Comment (yucaowang) [inline cmd/cli/account_utxo_records.go]: Do we have default value for address? could be the address at ./data/keys/address?

File: utxo/utxo.go (hunk 4 above)
Comment (ToWorld) [reply utxo/utxo.go]: The doc of goleveldb says that the returned DB instance is safe for concurrent use, meaning that all DB's methods may be called concurrently from multiple goroutine.

**Pull request merged**